include/istio/control/http/check_data.h

```diff
@@ -89,6 +89,12 @@ class CheckData {
   // If the request has authentication result in header, parses data into the
   // output result; returns true if success. Otherwise, returns false.
   virtual bool GetAuthenticationResult(istio::authn::Result *result) const = 0;
+
+  // Get request url path, which strips query part from the http path header.
+  virtual std::string GetUrlPath() const = 0;
```

Comment:
Please following the pattern of other methods; return bool to indicate if there is such value.

Reply:
Done.